source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -103,12 +103,13 @@ void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callb
   decoding_state_.setDecoderFilterCallbacks(callbacks);
   const Envoy::StreamInfo::FilterStateSharedPtr& filter_state =
       callbacks.streamInfo().filterState();
-  if (!filter_state->hasData<ExtProcLoggingInfo>(ExtProcLoggingInfoName)) {
-    filter_state->setData(ExtProcLoggingInfoName, std::make_shared<ExtProcLoggingInfo>(),
+  if (!filter_state->hasData<ExtProcLoggingInfo>(callbacks.filterConfigName())) {
+    filter_state->setData(callbacks.filterConfigName(),
+                          std::make_shared<ExtProcLoggingInfo>(config_->metadata()),
                           Envoy::StreamInfo::FilterState::StateType::Mutable,
                           Envoy::StreamInfo::FilterState::LifeSpan::Request);
   }
-  logging_info_ = filter_state->getDataMutable<ExtProcLoggingInfo>(ExtProcLoggingInfoName);
+  logging_info_ = filter_state->getDataMutable<ExtProcLoggingInfo>(callbacks.filterConfigName());
```

Comment:
I think we should have this qualified. What I would suggest is that we prefix this with something like `envoy.filter.ext_proc.<config name>`. Otherwise, we might end up conflicting with some random part of the filer namespace.

@adisuissa we should have better guidance here I think, re: my comment on filter state name space.

Reply:
I thought about this. Given that per-route configuration is also using the config name and this kind of conflict doesnt happen there as the filter names are required to be unique for the per-route config to work, I felt that using the same config name here wont collide. I dont mind changing if this assumption is incorrect

Reply:
Instead of adding filter namespace as prefix, which could become `envoy.filter.ext_proc.envoy.filter.ext_proc` if people set filter name as `envoy.filter.ext_proc`,

can we  add a suffix like `.logging_info` which becomes `envoy.filter.ext_proc.logging_info` or `envoy.filter.ext_proc_1.logging_info` This will be a clear key for the type of value it store and It should be good as along as it is not same as filter namsapce?

Reply:
Harvey's point is about key collisions. If the filter namespace is same as filter config name which is unique, then key collisions dont happen. The suffix piece is an independent conversation I feel. While it might signify that the filter state contains stats, it has an unintended side effect: too many filter states from same filter. Our original intent when designing the filter state years ago was to make this as a way for filters to communicate typed data. So all things shared by a filter are under one stable key that other filters can use as they wish.

Reply:
Yeah, the per-route use of non-prefixed name is fair, it seems there is precedent to use direct naming. No objection then to how it is done in this PR. I do think it behooves us to do a better job documenting how filter state is intended to be used. I do think we had some ideas years ago and some of them stuck and others are /shrug.

Reply:
What I suggested is still one per filter, same as what you have here but with the additional suffix `logging_info`.  Essentially it can be any string as it is just the key of the `data_storage_` map in filter state. 

I agree it is an independent topic and I don't feel strongly that we need to do this here. We can add suffix later if we want.